habitat/datasets/rearrange/samplers.py

```diff
@@ -417,26 +433,48 @@ def sample(
             and num_pairing_tries < self.max_sample_attempts
         ):
             num_pairing_tries += 1
-            new_object = self.single_sample(sim, snap_down, vdb)
-            if new_object is not None:
-                targets_found += 1
-                found_match = False
+
+            new_object, receptacle = self.single_sample(sim, snap_down, vdb)
+            found_match = False
+            for _ in range(1000):
+                if found_match:
+                    break
+                new_object, receptacle = self.single_sample(
+                    sim, snap_down, vdb
+                )
+                if new_object is None:
+                    continue
+
                 for object_instance in self.object_instance_set:
                     if (
                         object_instance.creation_attributes.handle
                         == new_object.creation_attributes.handle
                         and object_instance.handle not in new_target_objects
+                        and target_receptacle is not None
+                        and object_to_containing_receptacle is not None
+                        and object_to_containing_receptacle[
+                            object_instance.handle
+                        ].parent_object_handle
+                        == target_receptacle.parent_object_handle
+                        and object_to_containing_receptacle[
+                            object_instance.handle
+                        ].parent_link
+                        == target_receptacle.parent_link
```

Comment:
Looks like this is restricting all instance sampling to only objects in the target_receptacle, right? Maybe this should be a specific derived sampler or configuration option instead of the only target generating option?